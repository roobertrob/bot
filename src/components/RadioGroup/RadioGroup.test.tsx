import '@testing-library/jest-dom';
import { getByText, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { RadioGroupComponent } from '.';

const options = [
  {
    id: 0,
    name: 'Tangran',
  },
  {
    id: 1,
    name: 'Price Action',
  },
];
const selected = {
  id: 1,
  name: 'Price Action',
};
const handleChangeSelected = jest.fn()

describe('Radio Group component', () => {
  it('should render Radio Group', () => {
    render(
      <RadioGroupComponent
        options={options}
        selected={selected}
        onChange={handleChangeSelected}
      />,
    );
    expect(screen.getByTestId('radio-group-component')).toMatchSnapshot();
  });
  it('should call function handleChangeSelected when user clicks', () => {
    render(
      <RadioGroupComponent
        options={options}
        selected={selected}
        onChange={handleChangeSelected}
      />,
    );
    userEvent.click(screen.getByText('Price Action'));
    expect(handleChangeSelected).not.toHaveBeenCalled();
    userEvent.click(screen.getByText('Tangran'));
    expect(handleChangeSelected).toHaveBeenCalled();
  });
});
