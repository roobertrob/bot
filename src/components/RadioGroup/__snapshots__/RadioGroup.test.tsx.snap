// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group component should render Radio Group 1`] = `
<div
  data-testid="radio-group-component"
  id="headlessui-radiogroup-:r0:"
  role="radiogroup"
>
  <div
    class="space-y-2"
    role="none"
  >
    <div
      aria-checked="false"
      aria-labelledby="headlessui-label-:r2:"
      class="border-transparent  bg-white border-gray2 relative block border rounded-sm shadow-sm px-6 py-4 cursor-pointer sm:flex sm:justify-between focus:outline-none h-[66px] hover:opacity-80"
      id="headlessui-radiogroup-option-:r1:"
      role="radio"
      tabindex="0"
    >
      <span
        class="flex items-center "
      >
        <span
          class="text-body2 text-body-2 text-sm flex flex-col"
        >
          <span
            class="font-medium"
            id="headlessui-label-:r2:"
          >
            Tangran
          </span>
        </span>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-labelledby="headlessui-label-:r4:"
      class="border-transparent  bg-white border-gray2 relative block border rounded-sm shadow-sm px-6 py-4 cursor-pointer sm:flex sm:justify-between focus:outline-none h-[66px] hover:opacity-80"
      id="headlessui-radiogroup-option-:r3:"
      role="radio"
      tabindex="-1"
    >
      <span
        class="flex items-center "
      >
        <span
          class="text-body2 text-body-2 text-sm flex flex-col"
        >
          <span
            class="font-medium"
            id="headlessui-label-:r4:"
          >
            Price Action
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
